addon/internal-session.js

```diff
@@ -110,7 +128,7 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     if (!Ember.isEmpty(this.authenticator)) {
       Ember.set(data, 'authenticated', Ember.merge({ authenticator: this.authenticator }, data.authenticated || {}));
     }
-    this.store.persist(data);
+    return this._callStoreAsync('persist', data);
   },
```

Comment:
`setUnknownProperty` returns the value that `set` returns usually (see https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/mixins/-proxy.js#L103) which is the value that has been set. I think that should be returned here as well as returning sth. else might lead to very hard to debug errors.